### PR TITLE
vtenv: Introduce vtenv for passing in collation & parser information

### PR DESCRIPTION
**Pull request by dbussink**
We have many places taking already 3 arguments for the collation env, parser & MySQL version. This is due to the refactors to pass in explicit state.

This introduces `vtenv` to refactor things a bit and slightly clean it up. We now have one thing that can carry all three dependencies and also potential future additional ones where we need it.

## Related Issue(s)

Part of #14717 

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported to release branches
-   [x] If this change is to be back-ported to previous releases, a justification is included in the PR description
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on CI?
-   [x] Documentation was added or is not required

### CODE REVIEW

File: go/vt/vtenv/vtenv.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	"vitess.io/vitess/go/mysql/config"
+	"vitess.io/vitess/go/vt/sqlparser"
+)
+
+type Environment struct {
+	collationEnv   *collations.Environment
+	parser         *sqlparser.Parser
+	mysqlVersion   string
+	truncateUILen  int
+	truncateErrLen int
```
Comment (arthurschreiber) [inline go/vt/vtenv/vtenv.go:14]: These two feel a bit out of place, because they seem very specific to the status pages. 🤷🏻‍♂️

**Review by arthurschreiber — APPROVED** [review]
This is a great cleanup! ❤️

**Review by ajm188 — APPROVED** [review]
i like this!

File: go/vt/vtenv/vtenv.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
+package vtenv
+
+import (
```
Comment (frouioui) [inline go/vt/vtenv/vtenv.go:3]: Header is missing here

**Comment by frouioui** [comment]
We need to re-run `make sizegen` as well, the static check workflow fails with `Regenerated cached_size files do not match the current version.`

**Review by frouioui — APPROVED** [review]
That's a really nice refactor, thank you!

File: go/vt/vtenv/vtenv.go (hunk 1 above)
Comment (dbussink) [reply go/vt/vtenv/vtenv.go:14]: They are used in a few more places but yeah. I don’t have a much better idea though, we need to pass them all the through then.

**Review by shlomi-noach — APPROVED** [review]

**Pull request merged**